hack/docgen/README.md

```diff
@@ -0,0 +1,29 @@
+# Cluster Monitoring Operator Docgen
+
+This package contains the code that automatically generates `Documentation/api.md` the code is heavily inspired by the (current) prometheus-operator codegen solution however, some adaptations were made to the format to alight it as much as possible with the OpenShift docs format.
+
+### Goal
+The goal of this code is to automate the generation of documentation that will not only be used by users but also by our technical writers who have to maintain an updated version of it on OpenShift docs. Because of this requirement, a custom code solution was used instead of an open source one to facilitate compliance with the OpenShift docs format. 
```

Comment:
```suggestion
The goal of this code is to automate the generation of documentation that will not only be used by users but also by our technical writers who have to maintain an updated version of it in the OpenShift documentation. Because of this requirement, a custom code solution was used instead of an open source one to facilitate compliance with the OpenShift docs format. 
```